airflow/templates.py

```diff
@@ -56,9 +56,11 @@ def ts_filter(value):
     return value.isoformat()
 
 
-def ts_nodash_filter(value):
-    return value.strftime('%Y%m%dT%H%M%S')
-
+def ts_nodash_filter(*value):
+    if value:
+      return value.strftime('%Y%m%dT%H%M%S')
+    else:
+      return None
```

Comment:
This is wrong since

1. `datetime(0, 0, 0)` is falsy.
2. The `*` should not exist.

Something like this should be implemented instead

```python
def ts_nodash_filter(value):
    if value is None:
        return None
    return value.strftime('%Y%m%dT%H%M%S')
```

Also please add the same logic to `ds_filter`, `ds_nodash_filter`, `ts_filter`, and `ts_nodash_with_tz_filter` as well.

Reply:
Search “falsy python” in any search engine.

> I thought if we take the value as variable args it would work since if an empty list is passed or with values is it not the case

It is not, the filter is passed one single value, the datetime, and we are adding None to the possible input type.

Reply:
Ok